src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs

```diff
@@ -768,10 +768,28 @@ private void RemoveDataCollectors(IReadOnlyCollection<DataCollectorInformation>
 
     private void LogAttachments(List<AttachmentSet> attachmentSets)
     {
+        if (attachmentSets is null)
+        {
+            EqtTrace.Error("DataCollectionManager.LogAttachments: Unexpected null attachmentSets.");
+            return;
+        }
+
         foreach (var entry in attachmentSets)
         {
+            if (entry is null)
+            {
+                EqtTrace.Error("DataCollectionManager.LogAttachments: Unexpected null entry inside attachmentSets.");
+                continue;
```

Comment:
Is it ok to only skip null parts or shall we assume something is wrong and fail? Same remarks for the attachments below.

Reply:
it's a log part...I don't want throw nothing if state is corrupted for some bug and we have enabled diagnostic information.